include/pagmo/algorithms/bee_colony.hpp

```diff
@@ -0,0 +1,369 @@
+            if (m_verbosity > 0u) {
+                // Every m_verbosity generations print a log line
+                if (gen % m_verbosity == 1u || m_verbosity == 1u) {
+                    auto best_idx = pop.best_idx();
+                    // Every 50 lines print the column names
+                    if (count % 50u == 1u) {
+                        print("\n", std::setw(7), "Gen:", std::setw(15), "Fevals:", std::setw(15), "Current best:",
+                              std::setw(15), "Best:\n");
+                    }
+                    print(std::setw(7), gen, std::setw(15), prob.get_fevals() - fevals0, std::setw(15),
+                          pop.get_f()[best_idx][0], std::setw(15), pop.champion_f()[0], '\n');
+                    ++count;
+                    // Logs
+                    m_log.push_back(
+                        log_line_type(gen, prob.get_fevals() - fevals0, pop.get_f()[best_idx][0], pop.champion_f()[0]));
+                }
+            }
+        }
+        return pop;
+    }
+
+    /// Sets the seed
+    /**
+     * @param seed the seed controlling the algorithm stochastic behaviour
+     */
+    void set_seed(unsigned int seed)
+    {
+        m_e.seed(seed);
+        m_seed = seed;
+    }
+    /// Gets the seed
+    /**
+     * @return the seed controlling the algorithm stochastic behaviour
+     */
+    unsigned int get_seed() const
+    {
+        return m_seed;
+    }
+    /// Sets the algorithm verbosity
+    /**
+     * Sets the verbosity level of the screen output and of the
+     * log returned by get_log(). \p level can be:
+     * - 0: no verbosity
+     * - >0: will print and log one line each \p level generations.
+     *
+     * Example (verbosity 100):
+     * @code{.unparsed}
+     *    Gen:        Fevals:  Current best:          Best:
+     *       1            100        81757.2        81757.2
+     *     101          10100        28.2248        28.2248
+     *     201          20100         7.9007       0.580009
+     *     301          30100        1.48624      0.0871484
+     *     401          40100       0.415268      0.0871484
+     *     501          50100       0.533034      0.0871484
+     * @endcode
+     * Gen is the generation number, Fevals the number of function evaluation used, Current best is the best fitness
+     * currently in the population, Best is the best fitness found.
+     *
+     * @param level verbosity level
+     */
+    void set_verbosity(unsigned int level)
+    {
+        m_verbosity = level;
+    }
+    /// Gets the verbosity level
+    /**
+     * @return the verbosity level
+     */
+    unsigned int get_verbosity() const
+    {
+        return m_verbosity;
+    }
+    /// Gets the number of generations
+    /**
+     * @return the number of generations to evolve for
+     */
+    unsigned int get_gen() const
+    {
+        return m_gen;
+    }
+    /// Algorithm name
+    /**
+     * One of the optional methods of any user-defined algorithm (UDA).
```

Comment:
I think we do not mention anymore this elsewhere. Can you remove? Maybe copy it from a recent algo ...